tests/src/test/java/io/streamnative/pulsar/handlers/kop/EntryPublishTimeKafkaFormatTest.java

```diff
@@ -0,0 +1,135 @@
+        long startTime = System.currentTimeMillis();
+
+        for (int i = 0; i < totalMsgs; i++) {
+            Thread.sleep(10);
+            String messageStr = messageStrPrefix + i;
+            kProducer.getProducer()
+                    .send(new ProducerRecord<>(
+                            topicName,
+                            0,
+                            System.currentTimeMillis(),
+                            i,
+                            messageStr))
+                    .get();
+            log.debug("Kafka Producer Sent message: ({}, {})", i, messageStr);
+        }
+
+        KConsumer kConsumer = new KConsumer(topicName, "localhost", getKafkaBrokerPort(), false,
+                null, null, "KafkaEntryConsumerGroup");
+        kConsumer.getConsumer().subscribe(Collections.singleton(topicName));
+
+        for (int i = 0; i < totalMsgs; i++) {
+            ConsumerRecords<Integer, String> records = kConsumer.getConsumer().poll(Duration.ofSeconds(1));
+            for (ConsumerRecord<Integer, String> record : records) {
+                assertTrue(record.timestamp() > startTime);
+                assertTrue(record.timestamp() < System.currentTimeMillis());
+            }
+        }
+
+        // time before first message
+        Map<TopicPartition, Long> targetTimes = Maps.newHashMap();
+        targetTimes.put(tp, startTime);
+
+        ListOffsetRequest.Builder builder = ListOffsetRequest.Builder
+                .forConsumer(true, IsolationLevel.READ_UNCOMMITTED)
+                .setTargetTimes(targetTimes);
+
+        KafkaCommandDecoder.KafkaHeaderAndRequest request = buildRequest(builder);
+        CompletableFuture<AbstractResponse> responseFuture = new CompletableFuture<>();
+        kafkaRequestHandler.handleListOffsetRequest(request, responseFuture);
+
+        AbstractResponse response = responseFuture.get();
+        ListOffsetResponse listOffsetResponse = (ListOffsetResponse) response;
+        assertEquals(listOffsetResponse.responseData().get(tp).error, Errors.NONE);
+        assertTrue(listOffsetResponse.responseData().get(tp).offset == 0);
+    }
+
+    KafkaCommandDecoder.KafkaHeaderAndRequest buildRequest(AbstractRequest.Builder builder) {
+        AbstractRequest request = builder.build();
+        builder.apiKey();
+
+        ByteBuffer serializedRequest = request
+                .serialize(new RequestHeader(builder.apiKey(), request.version(), "fake_client_id", 0));
+
+        ByteBuf byteBuf = Unpooled.copiedBuffer(serializedRequest);
+
+        RequestHeader header = RequestHeader.parse(serializedRequest);
+
+        ApiKeys apiKey = header.apiKey();
+        short apiVersion = header.apiVersion();
+        Struct struct = apiKey.parseRequest(apiVersion, serializedRequest);
+        AbstractRequest body = AbstractRequest.parseRequest(apiKey, apiVersion, struct);
+        return new KafkaCommandDecoder.KafkaHeaderAndRequest(header, body, byteBuf, serviceAddress);
+    }
+
+}
```

Comment:
Can you reuse the `KafkaApisTest`? I found it's the same with `KafkaApisTest#buildRequest`. Also I think the tests can be put into `KafkaApisTest` as well.

Reply:
> Also I think the tests can be put into KafkaApisTest as well.

Oh I just noticed that the `KafkaApiTest` only supports pulsar entry format.